### PR TITLE
Examples of using various automatic differentiation libraries. 

### PR DESCRIPTION
**Pull request by zhanggiene**
Demonstration the capability of using Autodiff and Stanmath Libraries. 
Design and implementation of 4 different pdf with various complexity and dimensions. 
Speed comparsion of different automatic differentiation libraires in the comtext of hmc sampling. 
Testing of Nutshmc with complex and high-dimensional pdf. 
Extend the polytope cube to length or more than 1.

### CODE REVIEW

File: examples/autodiff_library/Gaussian_mixture_Autodiff_backward.cpp [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+    std::cerr << "Average Acceptance Probability: " << exp(hmc.average_acceptance_log_prob) << std::endl;
+    std::cerr << "PSRF: " << multivariate_psrf<NT, VT, MT>(samples) << std::endl;
+    std::cerr << std::endl;
+}
+
+int main()
+{
+    run_main<double>();
+    return 0;
+}
```
Comment (vissarion) [inline examples/autodiff_library/Gaussian_mixture_Autodiff_backward.cpp:183]: please add newline at the end of file

File: examples/autodiff_library/Gaussian_mixture_Autodiff_forward.cpp [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+        NT kappa; // Condition number
+        Eigen::Matrix<autodiff::detail::Real<1, NT>, Eigen::Dynamic, Eigen::Dynamic> data;
+        parameters() : order(2), L(4), m(4), kappa(1)
+        {
+            data = readMatrix<NT>("data.txt");
+        };
+    };
+
+    template <typename NT>
+    static autodiff::detail::Real<1, NT> userDefinedFunction(const Eigen::Matrix<autodiff::detail::Real<1, NT>, Eigen::Dynamic, 1> &x, const Eigen::Matrix<autodiff::detail::Real<1, NT>, Eigen::Dynamic, Eigen::Dynamic> &data_)
```
Comment (vissarion) [inline examples/autodiff_library/Gaussian_mixture_Autodiff_forward.cpp:60]: too long line

File: examples/autodiff_library/Gaussian_mixture_Autodiff_backward.cpp (hunk 1 above)
Comment (vissarion) [reply examples/autodiff_library/Gaussian_mixture_Autodiff_backward.cpp:183]: same for many files in this PR

**Review by vissarion — APPROVED** [review]
LGTM, thanks!

**Comment by zhanggiene** [comment]
refer to #256 for an updated PR.

**Pull request closed without merging**